test/lazy/test_block_diag_lazy_tensor.py

```diff
@@ -67,6 +67,31 @@ def evaluate_lazy_tensor(self, lazy_tensor):
                     actual[i, k, j * 4 : (j + 1) * 4, j * 4 : (j + 1) * 4] = blocks[i, k, j]
         return actual
 
+class TestBlockDiagLazyTensorMetaClass(unittest.TestCase):
+    def test_metaclass_constructor(self):
+        k, n = 3, 5 # number of blocks, block size
```

Comment:
This only tests this without any batch dimension. Could you also add a unit test that checks this for the batched case?